chainapsis/keplr-wallet
# Extended Agoric Message Support

## Description

- Adds user-friendly rendering for "AgoricProvisionMessage".
- Extends rendering for "AgoricWalletSpendActionMessage" to support other asset types (see context below).

## Context

Currently, "AgoricWalletSpendActionMessage" only supports assets that are in the vbank, which are cosmos bank-level assets. There are other assets which will only exist on the smart contract level, and their display infos need to be queried separately from vbank assets. This PR adds those queries and updates the message renderer to fallback on them if the asset is not in the vbank.

## Testing

Using Agoric devnet:

- Create a new account on a Keplr build with these changes
- Copy your Agoric address into https://devnet.faucet.agoric.net/ as shown and submit the request to receive BLD tokens (not provision):
![faucet](https://github.com/chainapsis/keplr-wallet/assets/8848650/cf510e80-95af-4af2-88c7-bd14d07999c2)
- Go to the test dapp at https://64f978c872cc7a4243d3c8b9--dancing-cendol-66d789.netlify.app/ and connect your wallet, it should connect to Agoric devnet.
- After connecting, click the "Provision Smart Wallet" button
![provision](https://github.com/chainapsis/keplr-wallet/assets/8848650/f25b74a2-52d0-4a90-bece-88f74f015855)
- Observe that the message should render successfully in Keplr.
![provisionkeplr](https://github.com/chainapsis/keplr-wallet/assets/8848650/76c8ecc1-a46f-43d7-a253-c4252e6b204f)
- Approve the message in Keplr, then wait about 30 seconds. The button should say "Make Offer":
![makeoffer](https://github.com/chainapsis/keplr-wallet/assets/8848650/ef45b7d4-feed-40f0-ae0e-3113bc86eccd)
- Click the "Make Offer" button and approve the message in Keplr. Observe how the non-vbank assets "Places" are rendered successfully in Keplr:
![offerkeplr](https://github.com/chainapsis/keplr-wallet/assets/8848650/c88cfa00-9646-4888-829d-b06ff45cbb9e)
